### ISSUE TITLE
Hit `Invalid access token` when updating cocoon task in luci devicelab

### ISSUE THREAD

**Issue by keyonghan** [flutter/flutter#72457]
Failed build: https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20color_filter_and_fade_perf__e2e_summary/137/overview

Detailed:
```
Unhandled exception:
AuthenticatedClientError:
  URI: https://flutter-dashboard.appspot.com/api/update-task-status
  HTTP Status: 401
  Response body:
Invalid access token
#0      AuthenticatedCocoonClient.send (package:flutter_devicelab/framework/cocoon.dart:159:7)
<asynchronous suspension>
#1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:91:32)
<asynchronous suspension>
#2      retry (package:flutter_devicelab/framework/utils.dart:773:14)
<asynchronous suspension>
#3      Cocoon._sendCocoonRequest (package:flutter_devicelab/framework/cocoon.dart:115:31)
<asynchronous suspension>
#4      Cocoon.sendTaskResult (package:flutter_devicelab/framework/cocoon.dart:100:43)
<asynchronous suspension>
#5      _runTasks (file:///opt/s/w/ir/k/flutter/dev/devicelab/bin/run.dart:126:7)
<asynchronous suspension>
#6      main (file:///opt/s/w/ir/k/flutter/dev/devicelab/bin/run.dart:104:5)
<asynchronous suspension>
```

**Comment by keyonghan** [comment]
@CaseyHillers Could you please help take a look?

**Comment by CaseyHillers** [comment]
Swarming can only guarantee an access token is valid for 3 minutes (up to 45 minutes). This creates a race condition for the luci devicelab tasks, where retrying will most likely fix it.

We need to split uploading results from the test execution to prevent this issue.

**Comment by keyonghan** [comment]
Another two:
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20flutter_gallery__memory_nav/136/overview
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20flutter_gallery_android__compile/142/overview

**Comment by keyonghan** [comment]
This issue has been happening more frequently. With linux ones finish migration (by tomorrow), this will block the tree. Increasing the priority to p1.

**Comment by godofredoc** [comment]
Does this mean that uploading the metrics is taking more than 3 mins?

**Comment by CaseyHillers** [comment]
No. The issue is we request the token before running the test. The
devicelab tests run from 1 second to 15 minutes. After the test is
complete, the test runner uploads to Cocoon within 30 seconds.

This requires splitting the results uploading into a separate command in
the test runner and a separate step in the recipe.

**Comment by zanderso** [comment]
This is frequently resulting in tree closures so bumping to P0.

**Comment by CaseyHillers** [comment]
Once tree is reopened, https://github.com/flutter/flutter/pull/72765 can land to unblock the necessary recipe changes.

Recipe CL: https://flutter-review.googlesource.com/c/recipes/+/9483

**Comment by godofredoc** [comment]
Let's landed on red. This is actually the problem making the tree red.

**Comment by CaseyHillers** [comment]
All CLs have landed, and i'll monitor to see if this fixed the invalid tokens.

**Comment by CaseyHillers** [comment]
A new commit landed and I'm not seeing the upload failures. Please reopen if this happens again.

**Comment by CaseyHillers** [comment]
This occurred again.

@keyonghan do you think we can move the refresh-github-commits endpoint from a 15 minute cron job to run it at midnight PST every day instead?

This is a race condition as we have both ways to trigger commits

**Comment by keyonghan** [comment]
The new issue is not same as here. Here is the new one: https://github.com/flutter/flutter/issues/79408, which is same as https://github.com/flutter/flutter/issues/78841
